workflow/controller/operator.go

```diff
@@ -779,7 +787,8 @@ func (woc *wfOperationCtx) persistUpdates(ctx context.Context) {
 		woc.log.WithError(err).Warn("error updating taskset")
 	}
 
-	if woc.wf.Status.Phase.Completed() {
+	// Note that the woc.checkTaskResultsCompleted() resolves a ~1/30 garbage collection failure race conidtion.
```

Comment:
can we say in the comment "make sure the TaskResults are incorporated into WorkflowStatus before we delete them"?